openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -100,6 +100,17 @@ TCN_IMPLEMENT_CALL(jlong, SSLContext, make)(TCN_STDARGS, jint protocol, jint mod
 
     UNREFERENCED(o);
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
+    // TODO this is very hacky as io.netty.handler.ssl.OpenSsl#doesSupportProtocol also uses this method to test for supported protocols. Furthermore
```

Comment:
Currently with openssl 1.1.0 `io.netty.handler.ssl.OpenSsl#doesSupportProtocol` will just return incorrect results(all versions supported). I did not change anything in netty as I don't know a proper methods of checking supported protocols in openssl 1.1.0 and I think `javax.net.ssl.SSLEngine#getSupportedProtocols` (which uses `io.netty.handler.ssl.OpenSsl#doesSupportProtocol` in the netty implementation) has very very limited use cases. 
If we find a method of checking supported protocols in openssl 1.1.0 then to make the code cleaner, we should of course add a new native method (instead of using `SSLContext.make(int protocol, int mode)`) for this and then we need to change netty code.
Apache's tcnative does not support checking supported protocols (I think). So using `make()` for that was propably just a quick-fix and now openssl 1.1.0 forces us to use a proper way